daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/ElementCombinator1.scala

```diff
@@ -198,6 +197,14 @@ abstract class ElementParserBase(
         eAfterParser.get.parse1(pstate)
 
       if (pstate.processorStatus ne Success) return
+
+      val shouldValidate = erd.isSimpleType &&
+        pstate.dataProc.isDefined && pstate.dataProc.value.validationMode != ValidationMode.Off
```

Comment:
I think when "full" validation is on, we should not incur the overhead of 'limited' validation, we should leave it up to the Xerces validation at the end of whole parse.

So this should, I think, look for ValidationMode.Limited specifically.

Reply:
That's a change in behavior that might break some tests? We might have some tests that set validationMode=full/on but expect specific error messages from Daffodil's limited validation. But that might be considered a bug in those tests?

Note we also have a similar check that would need to be changed here: https://github.com/apache/daffodil/blob/main/daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/SequenceChildBases.scala#L532-L533. Might have others as well? So maybe if we want to change the behavior we might want to create a ticket and do it as as separate PR?

Reply:
Agreed. Separate ticket. Also current behaviour tests the facets for daffodil even when validation is full, and making sure the facets work in both can be important.